polaris-core/src/main/java/org/apache/polaris/core/persistence/PolarisMetaStoreManagerImpl.java

```diff
@@ -1502,6 +1506,26 @@ public Map<String, String> deserializeProperties(PolarisCallContext callCtx, Str
       }
     }
 
+    // Check that cleanup is enabled, if it is set:
+    if (catalogPath != null && cleanup) {
+      boolean dropWithPurgeEnabled =
+          callCtx
+              .getConfigurationStore()
+              .getConfiguration(
+                  callCtx,
+                  (CatalogEntity) catalogPath.get(0),
+                  PolarisConfiguration.DROP_WITH_PURGE_ENABLED);
+      if (dropWithPurgeEnabled) {
```

Comment:
Looks like this is inverted - should throw exception if `!dropWithPurgeEnabled`?

I wonder if we should use `BadRequestException` (which is sometimes thrown for other types of "violated constraints" that aren't necessarily controlled by RBAC) and reserve `ForbiddenException` for things that pertain to RBAC/grants

Reply:
There's a discussion to be had about https://github.com/apache/polaris/pull/264

My thinking has been that a request isn't inherently "bad" if setting a config can make it succeed. The RFC says of `BAD_REQUEST`:
> The client SHOULD NOT repeat the request without modifications.

On the other hand, `FORBIDDEN` has the connotation that in other contexts (or from other clients) the request might be "`ALLOWED`". This seems somewhat in line with the behavior we want, although it's not forbidden by RBAC it is forbidden due to a configuration.

In any event, I am okay switching to `BAD_REQUEST` here but I think we should try to be uniform in how Polaris responds to situations where a request is not possible due to a configuration.

Reply:
Yeah that's a good point, I'm okay with FORBIDDEN after re-reading the RFC.